docs/tutorials/data_management.md

```diff
@@ -178,6 +191,11 @@ impl_data_provider!(UnstableProvider);
 fn main() {
     let unstable_provider = UnstableProvider;
 
+    // Wrapping the raw UnstableProvider in a LocaleFallbackProvider enables
+    // locales to fall back to other locales, like "en-US" to "en".
+    let unstable_provider = LocaleFallbackProvider::try_new_unstable(unstable_provider)
```

Comment:
I'm trying this in my code but I get an error like this:

```
et provider = LocaleFallbackProvider::try_new_unstable(UnstableProvider)?;
    |                    ---------------------------------------- ^^^^^^^^^^^^^^^^ the trait `icu_provider::DataProvider<LocaleFallbackLikelySubtagsV1Marker>` is not implemented for `UnstableProvider`
    |                    |
    |                    required by a bound introduced by this call
    |
    = help: the following other types implement trait `icu_provider::DataProvider<M>`:
              <UnstableProvider as icu_provider::DataProvider<CanonicalDecompositionDataV1Marker>>
              <UnstableProvider as icu_provider::DataProvider<CanonicalDecompositionTablesV1Marker>>
              <UnstableProvider as icu_provider::DataProvider<CollationDataV1Marker>>
              <UnstableProvider as icu_provider::DataProvider<CollationDiacriticsV1Marker>>
              <UnstableProvider as icu_provider::DataProvider<CollationJamoV1Marker>>
              <UnstableProvider as icu_provider::DataProvider<CollationMetadataV1Marker>>
              <UnstableProvider as icu_provider::DataProvider<CollationReorderingV1Marker>>
              <UnstableProvider as icu_provider::DataProvider<CollationSpecialPrimariesV1Marker>>
note: required by a bound in `LocaleFallbackProvider::<P>::try_new_unstable`
   --> /home/autarch/.cargo/registry/src/github.com-1ecc6299db9ec823/icu_provider_adapters-1.2.0/src/fallback/adapter.rs:54:8
    |
54  |     P: DataProvider<LocaleFallbackLikelySubtagsV1Marker>
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `LocaleFallbackProvider::<P>::try_new_unstable`
```

Reply:
Ok, this is because I had to add more keys to the data generated by `icu4x-datagen`. The workflow around this is pretty frustrating:

1. Generate all the keys.
2. Build my program, which takes forever since I generated all the keys.
3. Generate using `--keys-for-bin`.
4. Add a new locale-related bit of code, go to step #1.

It'd be nice if there was more guidance on which keys are needed for each part of the icu crate's API.

Reply:
First, yes, acknowledged that this process is a bit frustrating. It was designed to give more control over the data lifecycle than ICU4C, and it goes very much to the extreme in that direction. For the next 1.3 release, we have https://github.com/unicode-org/icu4x/issues/2945 to streamline this for users who don't need all the control.

That said, there are ways you can speed up your development loop. For example, in step 1 and 2, if you generate data with `--locales none`, it builds a lot faster, and still works with `--keys-for-bin`.

> It'd be nice if there was more guidance on which keys are needed for each part of the icu crate's API.

You can see the list of keys in the signature of the `try_new_unstable` functions. For example: [`LineSegmenter::try_new_auto_unstable`](https://unicode-org.github.io/icu4x/docs/icu/segmenter/struct.LineSegmenter.html#method.try_new_auto_unstable) has the following bounds:

DataProvider<[LineBreakDataV1Marker](https://unicode-org.github.io/icu4x/docs/icu/segmenter/provider/struct.LineBreakDataV1Marker.html)> + DataProvider<[LstmForWordLineAutoV1Marker](https://unicode-org.github.io/icu4x/docs/icu/segmenter/provider/struct.LstmForWordLineAutoV1Marker.html)> + DataProvider<[GraphemeClusterBreakDataV1Marker](https://unicode-org.github.io/icu4x/docs/icu/segmenter/provider/struct.GraphemeClusterBreakDataV1Marker.html)>

If you click into those three data markers, it gives you the key in the docs string.